File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.java
import java.util.Arrays;
class Solution {
    public int[] solution(int[] arr) {
       int original = arr.length;
        int target = 1;
        while(target < original){
            target *= 2;
        }

        int[] answer = Arrays.copyOf(arr, target);

        return answer;
        
    }
}

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.js
function solution(arr) {
    let length = arr.length;

    // 필요한 길이 (2의 거듭제곱 중 가장 가까운 값)
    let nextPowerOfTwo = 1;
    while (nextPowerOfTwo < length) {
        nextPowerOfTwo *= 2;
    }

    // 부족한 만큼 0을 추가
    while (length < nextPowerOfTwo) {
        arr.push(0);
        length++;
    }

    return arr;
}
